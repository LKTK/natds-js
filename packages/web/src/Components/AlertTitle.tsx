import MaterialAlertTitle from '@material-ui/lab/AlertTitle';
export { AlertTitleProps as IAlertTitleProps } from '@material-ui/lab/AlertTitle';

export default MaterialAlertTitle;
