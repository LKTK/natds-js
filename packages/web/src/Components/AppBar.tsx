import React, { FunctionComponent, forwardRef } from 'react';
import MaterialAppBar, { AppBarProps } from '@material-ui/core/AppBar';
import { withTheme, makeStyles } from '@material-ui/core/styles';
import { IThemeWeb } from '../Themes';
import { getDefaultTheme } from './shared';

export interface IAppBarProps extends AppBarProps {
  /**
   * @optional
   */
  theme: IThemeWeb | unknown;
}

const defaultAppBarElevation = 2;

export const AppBar: FunctionComponent<IAppBarProps> = forwardRef((
  props: Omit<IAppBarProps, 'translate'>,
  ref: any
) => {
  const {
    elevation,
    theme: providerTheme,
    classes,
  } = props;

  const theme: any  = React.useMemo(() => getDefaultTheme(providerTheme), [providerTheme]);

  const useStyles = React.useMemo(() => makeStyles({
    colorDefault: {
      backgroundColor: theme.palette.background.paper,
      color: theme.palette.background.paperContrastText,
    }
  }), [theme]);

  const customClasses = useStyles();

  return <MaterialAppBar
    {...props}
    classes={{...customClasses, ...classes}}
    elevation={!elevation && elevation !== 0 ? defaultAppBarElevation : elevation}
    ref={ref}
  />;
});

export default withTheme(AppBar);
