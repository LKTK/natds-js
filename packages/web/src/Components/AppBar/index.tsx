import React, { FunctionComponent, forwardRef } from "react";
import MaterialAppBar, { AppBarProps } from "@material-ui/core/AppBar";
import { makeStyles, withTheme } from "@material-ui/core/styles";
import PropTypes from "prop-types";
import { IThemeWeb } from "../../Themes";
import { getDefaultTheme } from "../shared";

export const colors: { secondary: string; default: string; inherit: string; primary: string } = {
  default: "default",
  inherit: "inherit",
  primary: "primary",
  secondary: "secondary",
};

export const positions: { static: string; absolute: string; sticky: string; fixed: string; relative: string } = {
  absolute: "absolute",
  fixed: "fixed",
  relative: "relative",
  static: "static",
  sticky: "sticky",
};

export const optionsElevation: { min: number; max: number; range: boolean; step: number; value: number } = {
  range: true,
  min: 0,
  max: 24,
  step: 1,
  value: 2,
};

export interface IAppBarProps extends AppBarProps {

  /**
   * @optional
   */
  theme: IThemeWeb | unknown;
}

const defaultAppBarElevation = 2;

export const AppBar: FunctionComponent<IAppBarProps> = forwardRef((
  props: Omit<IAppBarProps, "translate">,
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  ref: any,
) => {
  const {
      elevation,
      theme: providerTheme,
      classes,
    } = props,

    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    theme: any = React.useMemo(() => getDefaultTheme(providerTheme), [providerTheme]),

    useStyles = React.useMemo(() => makeStyles({
      colorDefault: {
        backgroundColor: theme.palette.background.paper,
        color: theme.palette.background.paperContrastText,
      },
    }), [theme]),

    customClasses = useStyles();

  /**
   * @todo refactor(web): replace elevation magic number by corresponding token
   */
  const NO_ELEVATION = 0;

  return <MaterialAppBar
    {...props}
    classes={{
      ...customClasses,
      ...classes,
    }}
    elevation={!elevation && elevation !== NO_ELEVATION ? defaultAppBarElevation : elevation}
    ref={ref}
  />;
});

AppBar.displayName = "AppBar";

export default withTheme(AppBar);

AppBar.propTypes = {
  children: PropTypes.node,
};
