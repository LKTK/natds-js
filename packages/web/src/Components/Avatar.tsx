import React, { FunctionComponent, forwardRef } from "react";
import MaterialAvatar, { AvatarProps as MaterialAvatarProps } from "@material-ui/core/Avatar";
import { withTheme } from "@material-ui/styles";
import { IAvatarSizes } from "@naturacosmeticos/natds-styles";
import { IThemeWeb } from "Themes";

export type AvatarSizes = keyof IAvatarSizes;
export type AvatarColors = "primary" | "secondary" | "default" | string;

export interface IAvatarProps extends MaterialAvatarProps {

  /**
   * @optional
   * @default 'standard'
   * Size of the Avatar
   */
  size?: AvatarSizes;

  /**
   * @optional
   * @default 'primary'
   * Color of the Avatar
   */
  color?: AvatarColors;
  theme: IThemeWeb | unknown;
}

const withSizes = (size: AvatarSizes | undefined,
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  theme: any = {}) => {

  if (!size) {
    return {};
  }

  const { avatarSizes }: { avatarSizes: IAvatarSizes } = theme;

  if (!avatarSizes) {
    return {};
  }

  const value = avatarSizes[size],
    sizeInPx = `${value.size}px`;

  return {
    fontSize: value.fontSize,
    height: sizeInPx,
    width: sizeInPx,
  };
};

const withColors = (color: AvatarColors | undefined,
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  theme: any = {}) => {

  if (!color) {
    return {};
  }

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const { palette = {} }: { palette: any } = theme,

    selectedColor = palette[color];

  if (!selectedColor || color === "default") {
    return {};
  }

  return {
    background: selectedColor.main,
    color: selectedColor.contrastText,
  };
};

export const Avatar: FunctionComponent<IAvatarProps> = forwardRef((
  props: IAvatarProps,
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  ref: any,
) => {
  const style = {
    ...withSizes(props.size, props.theme),
    ...withColors(props.color, props.theme),
    ...props.style,
  };

  return <MaterialAvatar {...props} style={style} ref={ref} />;
});

Avatar.displayName = "Avatar";

export default withTheme(Avatar);
