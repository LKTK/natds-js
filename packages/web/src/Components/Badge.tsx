import React, { FunctionComponent, forwardRef } from "react";
import { withTheme } from "@material-ui/styles";
import MaterialBadge, { BadgeProps as IBadgeProps } from "@material-ui/core/Badge";

export { BadgeProps as IBadgeProps } from "@material-ui/core/Badge";

export const Badge: FunctionComponent<IBadgeProps> = forwardRef((
  props: IBadgeProps,
  ref: any,
) => <MaterialBadge {...props} ref={ref} />);

export default withTheme(Badge);
