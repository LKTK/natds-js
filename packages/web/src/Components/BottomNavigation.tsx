import React, { FunctionComponent, forwardRef } from "react";
import MaterialBottomNavigation, { BottomNavigationProps as IBottomNavigationProps } from "@material-ui/core/BottomNavigation";
import { withTheme } from "@material-ui/core/styles";

export { BottomNavigationProps as IBottomNavigationProps } from "@material-ui/core/BottomNavigation";

export const BottomNavigation: FunctionComponent<IBottomNavigationProps> = forwardRef((
  props: IBottomNavigationProps,
  ref: any,
) => <MaterialBottomNavigation {...props} ref={ref} />);

export default withTheme(BottomNavigation);
