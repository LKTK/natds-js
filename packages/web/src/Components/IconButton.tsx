import React, { FunctionComponent, forwardRef } from 'react';
import { withTheme } from '@material-ui/styles';
import MaterialIconButton, {
  IconButtonProps as IIconButtonProps
} from '@material-ui/core/IconButton';
export { IconButtonProps as IIconButtonProps } from '@material-ui/core/IconButton';

export const IconButton: FunctionComponent<IIconButtonProps> = forwardRef((
  props: IIconButtonProps,
  ref: any
) => <MaterialIconButton {...props} ref={ref} />);

export default withTheme(IconButton);
