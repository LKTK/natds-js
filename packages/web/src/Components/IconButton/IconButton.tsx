import * as React from "react";

import MaterialIconButton from "@material-ui/core/IconButton";
import { IIconButtonProps } from "./IconButton.props";

export { IIconButtonProps } from "./IconButton.props";

/**
 * For more advanced usages, check [Material UI IconButton docs](https://material-ui.com/components/buttons/#icon-buttons)
 *
 * ## Importing
 *
 * ```jsx
 * import { IconButton } from '@naturacosmeticos/natds-web';
 * ```
 */
export const IconButton = React.forwardRef<unknown, IIconButtonProps>(
  (props: IIconButtonProps,
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
    ref: any,
  ) => <MaterialIconButton {...props} ref={ref} />);

IconButton.displayName = "IconButton";

export default IconButton;
