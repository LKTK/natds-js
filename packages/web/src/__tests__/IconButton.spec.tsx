import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { shallow } from 'enzyme';
import { expect } from 'chai';
import { assert, spy } from 'sinon';

import { IconButton } from '../Components/IconButton';

const MockedIcon = () => <i />;

describe('IconButton component', () => {
  describe('rendering sizes', () => {
    test('should match snapshot - Size Small', () => {
      const component = renderer.create(<IconButton size="small"><MockedIcon /></IconButton>).toJSON();

      expect(component).matchSnapshot('Small IconButton snapshot');
    });

    test('should match snapshot - Size Medium', () => {
      const component = renderer.create(<IconButton size="medium"><MockedIcon /></IconButton>).toJSON();

      expect(component).matchSnapshot('Medium IconButton snapshot');
    });
  });

  describe('interaction', () => {
    test('should call onClick', () => {
      const mockOnClick = spy();
      const component = shallow(<IconButton onClick={mockOnClick}><MockedIcon /></IconButton>);

      component.simulate('click');

      assert.calledOnce(mockOnClick);
    });
  });
});
