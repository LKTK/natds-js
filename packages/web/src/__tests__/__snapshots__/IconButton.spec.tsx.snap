// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton component rendering sizes should match snapshot - Size Medium: Medium IconButton snapshot 1`] = `
<button
  className="MuiButtonBase-root MuiIconButton-root"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiIconButton-label"
  >
    <i />
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;

exports[`IconButton component rendering sizes should match snapshot - Size Small: Small IconButton snapshot 1`] = `
<button
  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiIconButton-label"
  >
    <i />
  </span>
</button>
`;
